sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2409,13 +2409,19 @@ static ur_result_t SetKernelParamsAndLaunch(
       property_list.push_back({UR_EXP_LAUNCH_PROPERTY_ID_COOPERATIVE,
                                launch_property_value_cooperative});
     }
-
+  }
+  if (WorkGroupMemorySize) {
+    property_list.push_back(
+        {UR_EXP_LAUNCH_PROPERTY_ID_WORK_GROUP_MEMORY, WorkGroupMemorySize});
+  }
+  if (!property_list.empty()) {
```

Comment:
I think this is getting ugly, but I'll let scheduler experts decide: @sergey-semenov , @KseniyaTikhomirova .

Reply:
Ping @sergey-semenov , @KseniyaTikhomirova ?